poh-bundle-win/oh.ps1

```diff
@@ -0,0 +1,844 @@
+		Read-Host;
+		exit 1
+	}
+}
+
+function config_database {
+	# Find a free TCP port to run MySQL starting from the default port
+	Write-Host "Looking for a free TCP port for MySQL database..."
+
+	while ( Test-NetConnection $script:MYSQL_SERVER -Port $MYSQL_PORT -InformationLevel Quiet ) {
+    		Write-Host "Testing TCP port $MYSQL_PORT...."
+        	$script:MYSQL_PORT++
+	}
+
+	Write-Host "Found TCP port $MYSQL_PORT!"
+
+	# Creating MySQL configuration
+	Write-Host "Generating MySQL config file..."
+	if ( Test-Path "$OH_PATH/etc/mysql/my.cnf" ) {
+		mv -Force $OH_PATH/etc/mysql/my.cnf $OH_PATH/etc/mysql/my.cnf.old
+	}
+	(Get-Content "$OH_PATH/etc/mysql/my.cnf.dist").replace("DICOM_SIZE","$DICOM_MAX_SIZE") | Set-Content "$OH_PATH/etc/mysql/my.cnf"
+	(Get-Content "$OH_PATH/etc/mysql/my.cnf").replace("OH_PATH_SUBSTITUTE","$OH_PATH") | Set-Content "$OH_PATH/etc/mysql/my.cnf"
+	(Get-Content "$OH_PATH/etc/mysql/my.cnf").replace("MYSQL_SERVER","$MYSQL_SERVER") | Set-Content "$OH_PATH/etc/mysql/my.cnf"
+	(Get-Content "$OH_PATH/etc/mysql/my.cnf").replace("MYSQL_PORT","$MYSQL_PORT") | Set-Content "$OH_PATH/etc/mysql/my.cnf"
+	(Get-Content "$OH_PATH/etc/mysql/my.cnf").replace("MYSQL_DISTRO","$MYSQL_DIR") | Set-Content "$OH_PATH/etc/mysql/my.cnf"
+	(Get-Content "$OH_PATH/etc/mysql/my.cnf").replace("DATA_DIR","$DATA_DIR") | Set-Content "$OH_PATH/etc/mysql/my.cnf"
+	(Get-Content "$OH_PATH/etc/mysql/my.cnf").replace("TMP_DIR","$TMP_DIR") | Set-Content "$OH_PATH/etc/mysql/my.cnf"
+	(Get-Content "$OH_PATH/etc/mysql/my.cnf").replace("LOG_DIR","$LOG_DIR") | Set-Content "$OH_PATH/etc/mysql/my.cnf"
+
+#	sed -e "s/DICOM_SIZE/$DICOM_MAX_SIZE/g" -e "s/OH_PATH_SUBSTITUTE/$OH_PATH_ESCAPED/g" -e "s/MYSQL_PORT/$MYSQL_PORT/" -e "s/MYSQL_DISTRO/$MYSQL_DIR/g" $OH_PATH/etc/mysql/my.cnf.dist > $OH_PATH/etc/mysql/my.cnf
+}
+
+function inizialize_database {
+	# Recreate directory structure
+	[System.IO.Directory]::CreateDirectory("$OH_PATH/$DATA_DIR") > $null
+	[System.IO.Directory]::CreateDirectory("$OH_PATH/$TMP_DIR") > $null
+	[System.IO.Directory]::CreateDirectory("$OH_PATH/$LOG_DIR") > $null
+	[System.IO.Directory]::CreateDirectory("$OH_PATH/$DICOM_DIR") > $null
+	[System.IO.Directory]::CreateDirectory("$OH_PATH/$BACKUP_DIR") > $null
+    # Inizialize MySQL
+	Write-Host "Initializing MySQL database on port $MYSQL_PORT..."
+	switch -Regex ( $MYSQL_DIR ) {
+		"mariadb" {
+	    	Start-Process -FilePath "$OH_PATH\$MYSQL_DIR\bin\mysql_install_db.exe" -ArgumentList ("--datadir=$OH_PATH\$DATA_DIR --password=$MYSQL_ROOT_PW") -Wait -NoNewWindow -RedirectStandardOutput "$LOG_DIR/$LOG_FILE" -RedirectStandardError "$LOG_DIR/$LOG_FILE_ERR"
+        	}
+		"mysql" {
+		    Start-Process "$OH_PATH\$MYSQL_DIR\bin\mysqld.exe" -ArgumentList ("--initialize-insecure --basedir=$OH_PATH\$MYSQL_DIR --datadir=$OH_PATH\$DATA_DIR") -Wait -NoNewWindow -RedirectStandardOutput "$LOG_DIR/$LOG_FILE" -RedirectStandardError "$LOG_DIR/$LOG_FILE_ERR"; break
+        }
+	}
+
+#	if ( $? -ne 0 ){
```

Comment:
btw what about all the commented out lines? Are those meant for devops/admin to tweak the script or can they be removed?

Reply:
The script still does not manage error exit code in the mysql functions. I left it there (bash way :-) ) so if anyone want's to give it a try...
Unfortunately the start-process function doesn't seem to handle this...

Reply:
Let's merge to close the Sprint, improvements are welcome with a new PR!